src/main/scala/com/raphtory/core/components/querytracker/QueryProgressTracker.scala

```diff
@@ -15,6 +15,65 @@ import org.apache.pulsar.client.api.Schema
 
 import scala.collection.mutable.ListBuffer
 
+
+/**
+  * {s}`QueryProgressTracker`
+  *  : {s}`QueryProgressTracker` is used to track the progress of Raphtory queries in terms of number of perspectives processed and duration taken to process each perspective.
+  *    Queries in Raphtory run as a series of {s}`Perspectives` which are graph views at specific timestamps and windows as the query progresses.
+  *    The progress tracker thus helps track query progress until the job is completed. Query types supported include {s}`PointQuery`, {s}`RangeQuery` and {s}`LiveQuery`
+  *
+  * ## Methods
+  *
+  *   {s}`getJobId(): String`
+  *    : Returns job identifier for the query
+  *
+  *   {s}`getLatestPerspectiveProcessed(): Perspective`
+  *    : Returns the latest {s}`Perspective` processed by the query
+  *
+  *   {s}`getPerspectivesProcessed(): List[Perspective]`
+  *    : Returns list of perspectives processed for the query till the current timestamp
+  *
+  *   {s}`getPerspectiveDurations(): List[Long]`
+  *    : Returns the time duration taken to process each of perspectives till the current timestamp
+  *
+  *   {s}`isJobDone(): Boolean`
+  *    : Checks if job is complete
+  *
+  *   {s}`waitForJob()`
+  *    : Polling mechanism to checks if the job is complete, repeats check after every second
+  *
+  *   {s}`handleMessage(msg: QueryManagement): Unit`
+  *    : Handles message to process the {s}`Perspective` received in case the query is in progress, or {s}`JobDone` if the query is complete
+  *
+  *   {s}`run(): Unit`
+  *    : Starts the query tracker pulsar consumer
+  *
+  *   {s}`stop(): Unit`
+  *    : Stops the query tracker consumer
+  *
+  *
+  * Example Usage:
+  *
+  * ```{code-block} scala
+  *
+  * import com.raphtory.core.deploy.Raphtory
+  * import com.raphtory.lotrtest.LOTRGraphBuilder
+  * import com.raphtory.core.components.spout.instance.ResourceSpout
+  * import com.raphtory.GraphState
+  * import com.raphtory.output.FileOutputFormat
+  *
+  * val customConfig = Map(("raphtory.pulsar.endpoint", "localhost:1234"))
+  * Raphtory.createClient("deployment123", customConfig)
+  * val graph = Raphtory.createGraph(ResourceSpout("resource"), LOTRGraphBuilder())
+  * val queryProgressTracker = graph.rangeQuery(GraphState(),FileOutputFormat("/test_dir"),1, 32674, 10000, List(500, 1000, 10000))
+  * val jobId                = queryProgressTracker.getJobId()
+  * Thread.sleep(2000)
+  * val perspectivesProcessed = queryProgressTracker.getPerspectivesProcessed()
+  * queryProgressTracker.waitForJob()
```

Comment:
This example does not look like it would actually work? What is the createClient call supposed to do? Is this up-to-date with the latest changes?

Reply:
Removed `createClient` since it's not directly useful for `QueryProgressTracker`.  I've restricted the example to creating graph, running query, using `QueryProgressTracker` for retrieving progress stats